crates/bevy_pbr/src/render/light.rs

```diff
@@ -1753,17 +1617,34 @@ pub fn queue_shadows(
             // NOTE: Lights with shadow mapping disabled will have no visible entities
             // so no meshes will be queued
             for entity in visible_entities.iter().copied() {
-                if let Ok(mesh_handle) = casting_meshes.get(entity) {
-                    if let Some(mesh) = render_meshes.get(mesh_handle) {
-                        let mut key =
-                            ShadowPipelineKey::from_primitive_topology(mesh.primitive_topology);
+                if let Ok((mesh_handle, material_handle)) = casting_meshes.get(entity) {
+                    if let (Some(mesh), Some(material)) = (
```

Comment:
_Not for this PR_, but now that we read the `material` here, perhaps we could also have a per-material flag to disable casting shadows?

Reply:
Oh, that's a pretty neat idea, I guess it would be less ECS friendly though, since now you'd need to query all material instead of just querying for NotShadowCaster.